File: 3_sdk/LMZX SDK 生产环境/LMZX_SDKDemo_OC/LMZX_SDKDemo_OC/Data/Model/LMHomeTypeModel.m
//
//  LMHomeTypeModel.m
//  LMZX_SDKDemo_OC
//
//  Created by yj on 2017/3/29.
//  Copyright © 2017年 99baozi. All rights reserved.
//

#import "LMHomeTypeModel.h"

@implementation LMHomeTypeModel
- (instancetype)initWithDict:(NSDictionary *)dict
{
    self = [super init];
    if (self) {
        [self setValuesForKeysWithDictionary:dict];
    }
    return self;
}

+ (instancetype)homeTypeModelWithDict:(NSDictionary *)dict {
    return [[self alloc] initWithDict:dict];
}

- (void)setFunctions:(NSArray *)functions {
    _functions = functions;
    NSMutableArray *arr = [NSMutableArray array];
    for (NSDictionary *dict in functions) {
        LMHomeTypefunction *model = [LMHomeTypefunction homeTypefunctionWithDict:dict];
        [arr addObject:model];
    }
    _functionModels = arr;
}

- (void)setValue:(id)value forUndefinedKey:(NSString *)key {}
@end


@implementation LMHomeTypefunction

- (instancetype)initWithDict:(NSDictionary *)dict
{
    self = [super init];
    if (self) {
        [self setValuesForKeysWithDictionary:dict];
    }
    return self;
}

+ (instancetype)homeTypefunctionWithDict:(NSDictionary *)dict {
    return [[self alloc] initWithDict:dict];
}

- (void)setValue:(id)value forUndefinedKey:(NSString *)key {}

@end

File: 3_sdk/蒲公英 swift/LMZX_SDKDemo_Swift/Data/Model/LMHomeTypeModel.swift
//
//  LMHomeTypeModel.swift
//  LMZX_SDKDemo_Swift
//
//  Created by yj on 2017/3/31.
//  Copyright © 2017年 99baozi. All rights reserved.
//

import UIKit

class LMFunctionModel: NSObject {
    var icon = ""
    
    var title = ""
    
    var type = ""
    
    override init() {
        
    }
    
    init(dict:[String: Any]) {
        super.init()
        setValuesForKeys(dict)
    }
    
    override func setValue(_ value: Any?, forUndefinedKey key: String) {
        
    }
    
}


class LMHomeTypeModel: NSObject {
    var groupTitle = ""
    
    var titleColor = ""
    
    var functions: [[String: Any]]? {
        
        didSet{
            if let functions = functions {
                for dict in functions {
                    
                    functionModels.append(LMFunctionModel(dict: dict))
                }
                
                
            }
            
            
            
        }
        
    }
    
    var functionModels = [LMFunctionModel]()
    
    
    override init() {
        
    }
    
    init(dict:[String: Any]) {
        super.init()
        setValuesForKeys(dict)
    }
    
    override func setValue(_ value: Any?, forUndefinedKey key: String) {
        
    }

}
